pycqed/measurement/hdf5_data.py

```diff
@@ -124,9 +125,135 @@ def encode_to_utf8(s):
     Required because h5py does not support python3 strings
     '''
     # converts byte type to string because of h5py datasaving
-    if type(s) == str:
+    if isinstance(s, str):
         s = s.encode('utf-8')
     # If it is an array of value decodes individual entries
-    elif type(s) == np.ndarray or list:
+    elif isinstance(s, (np.ndarray, list)):
         s = [s.encode('utf-8') for s in s]
     return s
+
+
+def write_dict_to_hdf5(data_dict: dict, entry_point):
+    """
+    Args:
+        data_dict (dict): dictionary to write to hdf5 file
+        entry_point (hdf5 group.file) : location in the nested hdf5 structure
+            where to write to.
+    """
```

Comment:
I'll ad a test and see what happens (and fix it if it breaks) but I think it should work ...